src/test/java/cat/udl/eps/softarch/geolearning/steps/CreateMatchResultStepDefs.java

```diff
@@ -0,0 +1,145 @@
+package cat.udl.eps.softarch.geolearning.steps;
+
+import cat.udl.eps.softarch.geolearning.domain.Match;
+import cat.udl.eps.softarch.geolearning.domain.MatchResult;
+import cat.udl.eps.softarch.geolearning.domain.Player;
+import cat.udl.eps.softarch.geolearning.repository.MatchRepository;
+import cat.udl.eps.softarch.geolearning.repository.MatchResultRepository;
+import cat.udl.eps.softarch.geolearning.repository.PlayerRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+public class CreateMatchResultStepDefs {
+
+    private Player player1;
+    private Match match;
+    private MatchResult matchResult;
+    public String newResourceUri;
+
+    @Autowired
+    private PlayerRepository playerRepository;
+
+    @Autowired
+    private MatchRepository matchRepository;
+
+    @Autowired
+    private MatchResultRepository matchResultRepository;
+
+    @Autowired
+    private StepDefs stepDefs;
+
+    @Given("There is a player {string} with password {string} and {string}")
+    public void thereIsAPlayerWithPasswordAnd(String username, String password, String email) {
+        if(!playerRepository.existsById(username))
+        {
+            player1 = new Player();
+            player1.setUsername(username);
+            player1.setPassword(password);
+            player1.setEmail(email);
+            player1.encodePassword();
+            playerRepository.save(player1);
+        }
+    }
+
+    @And("There is a match with id {int}")
+    public void thereIsAMatchWithId(Integer matchId) {
+        if(!matchRepository.existsById(matchId))
+        {
+            match = new Match();
+            match.setId(matchId);
+            matchRepository.save(match);
+        }
+    }
+
+    @Given("There is no registered matchResult for this match attached to the player")
+    public void thereIsNoRegisteredMatchResultForThisMatchAttachedToThePlayer() {
+        Assert.assertNull(matchResultRepository.findByMatchAndPlayer(match, player1));
+    }
+
+    @Given("I played the match and is created MatchResult with the result {int} and time {int}")
+    public void iPlayedTheMatchAndIsCreatedMatchResultWithTheResultAndTime(int result, int time) throws Throwable {
+        MatchResult matchResult = new MatchResult();
+        matchResult.setResult(result);
+        matchResult.setTime(time);
+        matchResult.setMatch(match);
+        matchResult.setPlayer(player1);
```

Comment:
If you want to send this properties, which is good, you have to annotate the properties with `@JsonIdentityReference(alwaysAsId = true)`. This will make these properties to be mapped as its URI identifier. Right now, these properties are being mapped with the whole object and it is not what spring expects in a relation attribute.